File: baekjoon/4358/Main.java
import java.io.*;
import java.util.*;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		int count = 0;

		StringTokenizer st = new StringTokenizer(br.readLine(), " ");
		
		HashMap<String, Integer> hash = new HashMap<>();

		while(true){
			String str = br.readLine();
			count = count + 1;
			if(str == null || str.equals("")){
				break;
			}
			
			hash.put(str, hash.getOrDefault(str,0)+1);
		}

		Object[] key = hash.keySet().toArray();
		Arrays.sort(key);

		for(int i=0; i<key.length; i++){
			double ans = ((double)(hash.get(key[i])) / (double)(count)) * 100;
			sb.append(key[i]).append(" ").append(String.format("%.4f", ans)).append('\n');
			
		}
		System.out.println(sb);
	
		
	}
}

File: baekjoon/4358/Main.py
import sys
input = sys.stdin.readline

tree = {}
count = 0

while True:
  str = input().rstrip()

  if not str:
    break
  
  if str not in tree:
    tree[str] =1
  else:
    tree[str] = tree[str] + 1
  
  count = count + 1

tree2 = list(tree.keys())
tree2.sort()

for i in tree2:
  print('%s %.4f' %(i, tree[i] / count * 100.0))


